Nrf91 board rename

ioannisg opened the pull request:
Renaming of nRF91 DK in the tree (board target names and documentation)

ioannisg committed:
boards: arm: nrf9160_pca10090: rename board to nrf9160dk_nrf9160

We rename the nRF91 Dev Kit board target (nrf9160_pca10090)
to nrf9160dk_nrf9160. We update all associated references
in the supportive documentation and all nRF9160-related
cofigurations and overlay files in the samples and tests
in the tree.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
boards: arm: nrf9160dk: add deprecation warning for old board names

Add a deprecation warning for nrf9160 DK when the old names
(nrf9160_pca10090, nrf52840_pca10090) are used instead of the
new board name. For nRF9160 SoC: do this for both secure and
non-secure board targets.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
doc: allow redirect old url for nrf9160 DK documentation page

We are renaming the nrf9160 DK documentation page, and due to
re-organizing the doc directory we need to allow redirecting
the old URLs for nrf916 DK board targets' documentation.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

anangl reviewed (commented):
One more renaming needed, in [nrf9160dk_nrf9160_partition_conf.dts](https://github.com/zephyrproject-rtos/zephyr/blob/1eb90b96fb8832896c3cb44bc0c36a45b2f65728/boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts#L8).

I think that you should also rename `nrf52840_pca10090` to `nrf9160dk_nrf52840` in the same PR.

ioannisg commented on boards/arm/nrf9160dk_nrf9160/doc/index.rst:
```diff
@@ -1,14 +1,15 @@
-The nRF9160 PCA10090 is a single-board development kit for evaluation
+The nRF9160 Development Kit (PCA10090) is a single-board development kit
+for evaluation
 and development on the nRF9160 SiP for LTE-M and NB-IoT. The
```
ok, will do :)

ioannisg commented on boards/arm/nrf9160dk_nrf9160/Kconfig:
```diff
@@ -1,12 +1,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_NRF52840_GPIO_RESET
 	bool "Use nRF52840 PCA10090 GPIO reset pin"
```
Agreed. I left it because I did not want to touch the nRF52840 on this PR, but i can change it

ioannisg commented:
> One more renaming needed, in [nrf9160dk_nrf9160_partition_conf.dts](https://github.com/zephyrproject-rtos/zephyr/blob/1eb90b96fb8832896c3cb44bc0c36a45b2f65728/boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts#L8).
> 
> I think that you should also rename `nrf52840_pca10090` to `nrf9160dk_nrf52840` in the same PR.

Will add the commits on top in this PR

ioannisg committed:
boards: arm: nrf52840_pca10090: rename board to nrf9160dk_nrf52840

We rename the nRF91 Dev Kit board target for the NRF52840
controller (nrf52840_pca10090) to nrf9160dk_nrf52840. We
update all associated references in the supportive
documentation and all nRF9160-related cofigurations and
overlay files in the samples and tests in the tree. We
also remove an un-referenced board image file that had
erroneously been part of the documentation of this platform.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@anangl pls, take another look

anangl commented:
@ioannisg Please, remove the file `boards/arm/nrf9160dk_nrf52840/doc/img/nrf52840_pca10056.jpg`. This is not an adequate picture for that board. It seems that this file has been added by mistake in [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/e521ee651f0f5cc9f75501774036ddd48312358e#diff-be100f076811812ed45e16ca44f1a113), as it is not referenced from documentation or any other place.

Also the order of commits should be corrected. The commit that renames nrf52840_pca10090 to nrf9160dk_nrf52840 should be placed before the one that adds the corresponding deprecation warning.

ioannisg commented:
> @ioannisg Please, remove the file `boards/arm/nrf9160dk_nrf52840/doc/img/nrf52840_pca10056.jpg`. This is not an adequate picture for that board. It seems that this file has been added by mistake in [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/e521ee651f0f5cc9f75501774036ddd48312358e#diff-be100f076811812ed45e16ca44f1a113), as it is not referenced from documentation or any other place.
> 
> Also the order of commits should be corrected. The commit that renames nrf52840_pca10090 to nrf9160dk_nrf52840 should be placed before the one that adds the corresponding deprecation warning.

Yeah, I was thinking the same about this picture when doing the renaming work. Will add a commit that removes it.

ioannisg commented:
@anangl comments addressed
